text/0000-new-buildpack-toml-keys.md

```diff
@@ -0,0 +1,88 @@
+# Meta
+[meta]: #meta
+- Name: New Buildpack Descriptor Keys
+- Start Date: 2020-12-12
+- Author(s): [@jkutner](@jkutner)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for new keys in the `buildpack.toml` file.
+
+# Definitions
+[definitions]: #definitions
+
+* _buildpack descriptor_ - The `buildpack.toml` file that describes a buildpack.
+
+# Motivation
+[motivation]: #motivation
+
+New optional fields in the `buildpack.toml` would allow us to introduce more information into the Buildpack Registry search API and the web interface.
+
+# What it is
+[what-it-is]: #what-it-is
+
+* Add `description` as a free text field
+* Add `licenses` table
+
+# How it Works
+[how-it-works]: #how-it-works
+
+See the spec changes below for full details.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- It introduces some none machine readable fields to an otherwise machine readable descriptor.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Do Nothing
+
+We would not be able to include this information in the Buildpack Registry.
+
+## Put the new keys in `package.toml`
+
+If we want to separate this information from the `buildpack.toml`, we could introduce it to `package.toml`, and only use it when creating a buildpackpage. This would be compatible with the Buildpack Registry because it only support buildpackages.
+
+## Put the new information in the Registry only
+
+This information could be stored in some other way, completely independent of each version (i.e. a name for the buildpack regardless of version). This could be stored in the registry index, or in some annex to it.
```

Comment:
I still think we could do this just in the registry, but you probably want at least license to be per version b/c OSS projects can change licenses.

Reply:
Including this information in `buildpack.toml` and the derived labels makes a lot of sense to me, even in a world where the registry does not exist. It is analogous to the description field in the builder metadata and the `licenses` table in `project.toml`.

Come to think of it... `project.toml` could probably benefit from a description too...